source/code/plugins/out_oms.rb

```diff
@@ -100,12 +111,12 @@ def write(chunk)
         raise OMS::RetryRequestException, 'Missing configuration. Make sure to onboard.'
       end
 
-      # Group records based on their datatype because OMS does not support a single request with multiple datatypes. 
+      # Group records based on their datatype because OMS does not support a single request with multiple datatypes.
       datatypes = {}
       unmergable_records = []
       chunk.msgpack_each {|(tag, record)|
         if record.has_key?('DataType') and record.has_key?('IPName')
-          key = "#{record['DataType']}.#{record['IPName']}".upcase
```

Comment:
I would like to remove upcase here.it will save us additional allocation.
For example if omsagent receive 1000 syslog events, we will allocate 2000 strings at this line.

Reply:
@abenbachir That's not a good idea. The way that batches are created depends on the key (events with same key are batched together). Some events have `DataType`/`IPName` with inconsistent casing. If you want to avoid `.upcase` a prerequisite would be correcting all those values so that they are consistent.

Reply:
DataType and IPName were set in filters, this one in filter_syslog, and I can see others in different filters:
```
wrapper = {
          "DataType"=>"LINUX_SYSLOGS_BLOB",
          "IPName"=>"logmanagement",
          "DataItems"=>[record]
      }
```
We are actually using static string, we could make them all upercase at the source. "IPName"=>"LOGMANAGEMENT",

Reply:
As long as you make changes in all locations that IPName/DataType are used, sounds good. You will find they are present in more than just filter files; please search across all of source/code/plugins/* for instances of IPName/DataType.

Reply:
@hestolz this may be done on different PR